go/libkb/shared_dh.go

```diff
@@ -116,21 +173,37 @@ func (s *SharedDHKeyring) Sync(ctx context.Context) (err error) {
 	return nil
 }
 
+func (s *SharedDHKeyring) getUPAK(ctx context.Context, lctx LoginContext, me *User) (*keybase1.UserPlusAllKeys, error) {
+	if me != nil {
```

Comment:
That's fine for `SharedDHKeyring`. But `*User` is already passed through `SignupEngine->PaperKeyPrimary->PaperKeyGen->Delegator`. In other words, `Delegator` needs a `*User` from `PaperKeyGen`. So this change is just moving the `ExportToUserPlusAllKeys` up one step to `PaperKeyGen`. Did that make sense?